tests/storage/test_user_directory.py

```diff
@@ -504,6 +504,139 @@ def test_search_user_dir_start_of_user_id(self) -> None:
             {"user_id": BELA, "display_name": "Bela", "avatar_url": None},
         )
 
+    @override_config({"user_directory": {"search_all_users": True}})
+    def test_search_user_dir_ascii_case_insensitivity(self) -> None:
+        """Tests that a user can look up another user by searching for their name in a
+        different case.
+        """
+        CHARLIE = "@someuser:example.org"
+        self.get_success(
+            self.store.update_profile_in_user_dir(CHARLIE, "Charlie", None)
+        )
+
+        r = self.get_success(self.store.search_user_dir(ALICE, "cHARLIE", 10))
+        self.assertFalse(r["limited"])
+        self.assertEqual(1, len(r["results"]))
+        self.assertDictEqual(
+            r["results"][0],
+            {"user_id": CHARLIE, "display_name": "Charlie", "avatar_url": None},
+        )
+
+    @override_config({"user_directory": {"search_all_users": True}})
+    def test_search_user_dir_unicode_case_insensitivity(self) -> None:
+        """Tests that a user can look up another user by searching for their name in a
+        different case.
+        """
+        IVAN = "@someuser:example.org"
+        self.get_success(self.store.update_profile_in_user_dir(IVAN, "Иван", None))
+
+        r = self.get_success(self.store.search_user_dir(ALICE, "иВАН", 10))
+        self.assertFalse(r["limited"])
+        self.assertEqual(1, len(r["results"]))
+        self.assertDictEqual(
+            r["results"][0],
+            {"user_id": IVAN, "display_name": "Иван", "avatar_url": None},
+        )
+
+    @override_config({"user_directory": {"search_all_users": True}})
+    def test_search_user_dir_dotted_dotless_i_case_insensitivity(self) -> None:
+        """Tests that a user can look up another user by searching for their name in a
+        different case, when their name contains dotted or dotless "i"s.
+
+        Some languages have dotted and dotless versions of "i", which are considered to
+        be different letters: i <-> İ, ı <-> I. To make things difficult, they reuse the
+        ASCII "i" and "I" characters, despite having different lowercase / uppercase
```

Comment:
> they reuse the ASCII "i" and "I" characters

Should this be `they reuse the ASCII "i" and "I" glyphs`?

Reply:
Aren't glyphs something to do with fonts and text rendering?
I'm using "character" and "code point" interchangeably here.

Reply:
Oh, I see. In that case, I think this is fine as it is.

<details><summary>:warning: Pedantry Alert :warning: </summary>

I think the Unicode consortium agrees with you. From p15 of https://www.unicode.org/versions/Unicode15.0.0/ch02.pdf under "Characters, Not Glyphs":

> Characters are the abstract representations of the smallest components of written language that have
semantic value.

> Characters are represented by code points that reside only in a memory representation, as
strings in memory, on disk, or in data transmission. The Unicode Standard deals only with
character codes.

Though I usually think of `character` to mean `grapheme cluster`. From p11 of that same document:

> Therefore, the text elements encountered in most text pro-
cesses are represented as sequences of character codes. [...] Certain text elements correspond to what users
perceive as single characters. These are called grapheme clusters.

Given the confusion I try to avoid using the word "character" altogether.
</details>

Reply:
Let's switch this to "code point" then.